### PR TITLE
add type for createCollections

### LINKED ISSUE #14279: Type definition for Connection.prototype.createCollections() not in connections.d.ts
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the bug has not already been reported

### Mongoose version

8.1.0

### Node.js version

n/a

### MongoDB server version

n/a

### Typescript version (if applicable)

5.x

### Description

[Connection.prototype.createCollections()] (https://mongoosejs.com/docs/api/connection.html#Connection.prototype.createCollections()), isn't visible to Typescript. The definition wasn't added to connection.d.ts in the PR for the feature: https://github.com/Automattic/mongoose/pull/13324.

### Steps to Reproduce

Install latest version of mongoose (8.1.0), see that the connection.createCollections isn't in the type definition. 
image
Confirm by visiting the source code: https://github.com/Automattic/mongoose/blob/master/types/connection.d.ts

### Expected Behavior

_No response_

### PR DESCRIPTION
**Pull request by IslandRhythms**
closes #14279

### CODE REVIEW

File: test/types/connection.test.ts [hunk 1]
```diff
@@ -20,6 +20,8 @@ conn.readyState === 99;
 
 expectError(conn.readyState = 0);
 
+expectError(conn.createCollections());
```
Comment (vkarpov15) [inline test/types/connection.test.ts:23]: Why is this expectError here?

**Review by vkarpov15 — CHANGES_REQUESTED** [review]
Remove the expectError and should be good

File: test/types/connection.test.ts (hunk 1 above)
Comment (IslandRhythms) [reply test/types/connection.test.ts]: You said to write a test showing it could be used.

Comment (vkarpov15) [reply test/types/connection.test.ts]: `expectError()` means you expect the code within to cause a TypeScript compiler error. The problem isn't the test, just the `expectError()`.

**Pull request merged**